### DISCUSSION TITLE
How to add user's info to the Authentication section?

### DISCUSSION THREAD

**Discussion by alabhyajindal** [supabase/supabase#7031] [Questions] [answered]
Hello everyone!

I am using Magic Link authentication in my app. After the user logs in, I redirect the user to a form where they can fill in additional account information. 
I tried using update() provided by supabase-js client and it works as expected, but I can't see the user_metadata stored by it anywhere on the Supabase dashboard. I can access it by logging the user data returned by supabase.auth.user() however.

<img width="339" alt="image" src="https://user-images.githubusercontent.com/52493077/170715711-5ba8d4e2-eb75-438f-aba6-9b853bfa3d24.png">

I wanted to know how we can store this additional information in the Authentication -> Users section so all user's data is in one place.

![Screenshot 2022-05-27 194051](https://user-images.githubusercontent.com/52493077/170716733-a0d8e23e-ce0f-4a4a-82cc-90124fbae024.jpg)

Thanks in advance for any help!

**Comment by GaryAustin1** [comment, marked as answer]
That screen is just data SB has chosen to pull from auth.users table and possibly one other auth table and present.  Your metadata is visible in auth.users table if you use the table UI.  

You would need to create your own tool if you want to present and manage if you need more than Users page provides or using the Table UI on auth.users provides.

**Reply by alabhyajindal** [reply]
Thanks a lot! I didn't see the option to switch schema until now. It's perfect, thanks again.

<img width="960" alt="image" src="https://user-images.githubusercontent.com/52493077/170733804-8e3f7ca9-a5db-43de-9313-3deb36db4c15.png">

**Reply by anastely** [reply]
@GaryAustin1  Can I create a new table for users, called "clients"? After a user signs up, can I take their user ID and add it to the client's table along with other information like their first name, last name, partner_id, and so on?

**Reply by GaryAustin1** [reply]
@anastely that would normally be done like the user management guides show in the docs.  If you look in the Dashboard SQL editor under Quickstarts there is also a user management template for creating a trigger function on auth.users and inserting the data into your own profile table.

**Reply by anastely** [reply]
@GaryAustin1 Yeah i checked it thanks, but i got this error while using GUI to set the email foreign key


`Failed to update column "email": Failed to run SQL query: there is no unique constraint matching given keys for referenced table "users"`


<img width="693" alt="Screenshot 2023-09-20 at 5 29 29 PM" src="https://github.com/supabase/supabase/assets/24620979/34b78c9c-d2aa-4f5e-9b1d-82ea44dd29ad">

<img width="670" alt="Screenshot 2023-09-20 at 5 31 11 PM" src="https://github.com/supabase/supabase/assets/24620979/f525afda-2632-419f-a38c-537017f6bdf4">

**Reply by GaryAustin1** [reply]
You can't foreign key link to auth.users email as it is not a unique column anymore.  You don't really need the fk though for any of the columns to work unless you want to do cascade delete (which could also be done by a delete trigger).

**Reply by anastely** [reply]
The main reason is i want to take the email from the auth.users and link it in the public profiles table under the email column after the user sign up, so I thought that's how it is done, how should I handle this case?

**Reply by GaryAustin1** [reply]
You can put it there.  You just can't have an fk on it.  You just insert it in the auth.users insert trigger like the id.  It will be added.

**Reply by anastely** [reply]
thank you so much, I should read more about Postgres :D
